homework/index.js

```diff
@@ -29,18 +33,218 @@
+      class: 'widget-icon fas fa-clock fa-3x px-3 py-1',
+    });
+    createAndAppend('h4', divWidgetRepoUpdatedData, {
+      text: 'Updated:',
+      class: 'py-2 px-2 widget-title',
+    });
+    const updatedDate = new Date(repositoriesData.updated_at);
+    createAndAppend('h4', divWidgetRepoUpdatedData, {
+      text: updatedDate.toDateString(),
+      class: 'widget-value lead',
+    });
+    //
+    const divWidgetForkedNumbers = createAndAppend('div', leftColumn, {
+      class:
+        'd-flex flex-column  alert alert-primary justify-content-start align-items-center flex-sm-column flex-lg-row',
+    });
+    createAndAppend('i', divWidgetForkedNumbers, {
+      class: 'widget-icon fas fa-code-branch fa-3x px-3 py-1',
+    });
+    createAndAppend('h4', divWidgetForkedNumbers, {
+      text: 'Forks:',
+      class: 'py-2 px-2 widget-title',
+    });
+    createAndAppend('h4', divWidgetForkedNumbers, {
+      text: repositoriesData.forks,
+      class: 'widget-value lead',
+    });
+  };
+  // append fetched repositories information to select list/box
+  const appendRepositoriesToSelect = (repositoriesData, cb) => {
+    const selectList = document.getElementById('repo-select');
+    repositoriesData
+      .sort((a, b) => a.name.localeCompare(b.name))
+      .forEach((repository, index) => {
+        createAndAppend('option', selectList, {
+          text: repository.name,
+          value: index,
+        });
+      });
+    // create an eventlistener for select list options
+    selectList.addEventListener('change', () => {
+      selectList.parentElement.nextElementSibling.innerHTML = '';
+      cb(repositoriesData[selectList.value], repositoriesData[selectList.value].contributors_url);
+    });
+  };
+  // fetch repositories and contributions data as a callback hell so that once repositories data is fetched first, then the contribution data will be fetched.
+  const fetchRepositoriesAndContributions = url => {
+    fetchJSON(url, (errRepository, repositoriesData) => {
+      if (errRepository) {
+        createAndAppend('div', root, {
+          text: errRepository.message,
+          class: 'alert alert-danger display-3 text-center',
+        });
         return;
       }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
+      appendRepositoriesToSelect(repositoriesData, (repositoryObj, contributionsURL) => {
+        createRepositoryLayout(repositoryObj);
+        fetchJSON(contributionsURL, (errContributions, contributionsObj) => {
+          if (errContributions) {
+            createAndAppend('div', root, {
+              text: `${errContributions.message}... The data cannot be fetched`,
+              class: 'alert alert-danger display-3 text-center',
+            });
+            return;
+          }
+          createContributionsLayout(contributionsObj);
+        });
+      });
+    });
+  };
+  // create page layout under the root div
+  const buildWebPage = url => {
```

Comment:
If you take out the function call at the end (see my comment further below) this function just deals with creating the header. So `createHeader` or something similar would seem a more accurate name.

Reply:
I corrected it.